src/metadata.rs

```diff
@@ -59,6 +60,7 @@ pub struct Package {
   pub targets: Vec<Target>,
   pub features: HashMap<String, Vec<FeatureOrDependency>>,
   pub manifest_path: String,
+  pub checkout_root: Option<String>,
```

Comment:
This going to be a really frustrating comment -- apologies in advance.

I _really_ don't want to touch any of these `Metadata` structs if we can avoid it -- they're basically identical to the ones `cargo metadata` is emitting. Modifying them here means that we'll somehow have to synthesize additional data when generating from a `cargo metadata` invocation, and it blurs the line between generation of Metadata, and actual processing of that data.

Is it possible to defer calculating this until planning?

Its my fault that this is unclear -- I should put comments on these structs explaining how they were derived (e.g. experimentally from the output of `cargo metadata`).

Reply:
Not frustrating, no worries. I was just trying to locate the data near where it was created (i.e. not trying to re-engineer logic to build `cargo_config`). I can regenerate the `checkout_root` as long as the `SourceId` is preserved and I know what the git checkout root is for Cargo.

The other thing to keep in mind, depending on how purist you're trying to be with matching `cargo metadata` output, is that I'm technically modifying the `SourceId` to serialize across to planning.